### DISCUSSION TITLE
Using other parameters to auth

### DISCUSSION THREAD

**Discussion by TheRafaelFarias** [adonisjs/core#1382] [Help]
Im trying to use the auth, but need to use password parameter, and that not is what i want

In auth.ts the user provider needs password, how i can change? I create a new provider?

**Comment by TheRafaelFarias** [comment]
![Screenshot_20200802_175647](https://user-images.githubusercontent.com/44478887/89132319-ae012300-d4e9-11ea-9715-24cbc117f1e3.png)

**Comment by KararTY** [comment]
They have to be there and exist in your User Model, but you do not have to use them and you can `serializeAs: null`.
![image](https://user-images.githubusercontent.com/13697809/89194218-5fb25980-d5a7-11ea-9c81-faf68b95ba82.png)

**Comment by TheRafaelFarias** [comment]
I think don't explained so good, i have the Project model, and to login will have name and key (uuid). But in the auth contract, need to have the password

**Comment by TheRafaelFarias** [comment]
![image](https://user-images.githubusercontent.com/44478887/89208964-c463b900-d593-11ea-8e72-b9e742b1a894.png)

That is what i wan't to change

**Comment by KararTY** [comment]
You can make custom authorization logic, but the models will still require the two fields (`password`, `rememberMeToken`) even if you don't plan on using them.

(Per: https://github.com/adonisjs/core/discussions/1306#discussioncomment-41118)
```js
import Hash from '@ioc:Adonis/Core/Hash'

...

// Assuming providedName is a username-like (string) field.
// Assuming providedUID is a password-like (string) field.

const project = await Project.query().where({ name: providedName }).firstOrFail()

const isValid = await Hash.verify(project.key, providedUID)

if (!isValid) {
  // return error
}

await auth.login(project)
```
or if your key isn't hashed:
```js
// Assuming providedName is a username-like (string) field.
// Assuming providedUID is a string field.

const project = await Project.query().where({ name: providedName, key: providedUID }).firstOrFail()

await auth.login(project)
```

**Reply by TheRafaelFarias** [reply]
don't have how i change password to key?

Because is an api to my clients projects, and to work i wan't to the backend login with my dashboard
